### PR TITLE
docs: add SPM backend page

### PR DESCRIPTION
**Pull request by kattouf**
1. Added document to SPM backend
2. Fix backend src path in dock: `src/forge` -> `src/backend`

### CODE REVIEW

**Comment by kattouf** [comment]
@pepicrft I think you'll like it

File: docs/dev-tools/backends/spm.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# SPM Backend <Badge type="warning" text="experimental" />
+
+You may install executables managed by [Swift Package Manager](https://www.swift.org/documentation/package-manager) directly from Github releases.
+
+The code for this is inside of the mise repository at [`./src/backend/spm.rs`](https://github.com/jdx/mise/blob/main/src/backend/spm.rs).
+
+## Dependencies
+
+This relies on having `swift` installed. You can install it according to the [instructions](https://www.swift.org/install).
```
Comment (pepicrft) [inline docs/dev-tools/backends/spm.md:9]:
```suggestion
This relies on having `swift` installed. You can install it according to the [instructions](https://www.swift.org/install).

> [!NOTE]
> If you have Xcode installed and selected in your system via `xcode-select`, Swift is already available through the toolchain embedded in the Xcode installation.
```

**Review by pepicrft — APPROVED** [review]
Love it! Amazing job

File: docs/dev-tools/backends/spm.md (hunk 1 above)
Comment (kattouf) [reply docs/dev-tools/backends/spm.md:9]: Double-checked that note syntax supported https://vitepress.dev/guide/markdown#github-flavored-alerts

**Pull request merged**